src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -702,7 +702,7 @@ private void AddSegment(int sizeHint = 0)
             {
                 // Get a new buffer using the minimum segment size, unless the size hint is larger than a single segment.
                 // Also, the size cannot be larger than the MaxBufferSize of the MemoryPool
-                var owner = _memoryPool.Rent(Math.Min(sizeHint, _memoryPool.MaxBufferSize));
```

Comment:
revert?

Reply:
I figured I'd just stick it in this PR because it's not big enough for its own. Look at line 701.